automerge/README.md

```diff
@@ -0,0 +1,135 @@
+# Automerge
+
+The `promote` task initiates change by creating Pull Requests. Sometimes a team will want to merge these manually, and sometimes automation can assist. A development team may want changes to an early, shared 'dev' environment to be merged automatically, whereas changes into 'prod' will more likely require manual approval. This directory contains sample Tekton tasks that demonstrate ways to automatically merge Pull Requests. 
+
+We have two subdirectories: 'standalone' and 'webhooks'. In the former case, a Tekton PipelineRun must be created manually. In the latter, the [Tekton Dashboard Webhooks Extension](https://github.com/tektoncd/experimental/tree/master/webhooks-extension) is used to start an automerge PipelineRun in response to a PullRequest arriving at a GitOps repository.
+
+# Setup - both cases
+
+Our samples currently work with GitHub. They use the `hub` CLI to create a merge commit that when pushed, will merge the associated Pull Request. See [here](https://hub.github.com/hub-merge.1.html) for more details.
+
+## gitconfig
+
+Since we are creating git commits within the Tekton tasks, we need to know the username and email address to associate with them, as in our ['tekton-example'](../tekton-example/README.md). So, edit `gitconfig` and
+
+```sh
+kubectl create configmap promoteconfigmap --from-file=gitconfig
+```
+
+## Dockerfile
+
+The 'hub' CLI runs in a Docker container within a Tekton Pipeline. We've provided a sample Dockerfile. If you are taking the 'webhooks' path you can comment out the section that installs `yq`. Then, 
+
+```sh
+docker login
+docker build -t YOUR_DOCKER_HUB_ID/hub-test .
+docker push YOUR_DOCKER_HUB_ID/hub-test
+```
+
+## Create a Pull Request
+
+1. Fork our example gitops repository, https://github.com/rhd-gitops-example/gitops-example-dev. 
+1. Create a 'promotion' Pull Request either manually or using our ['tekton-example'](../tekton-example/README.md). For example you can use code of the form, 
+
+```sh
+services promote --from promote-demo --to https://github.com/YOUR_GITHUB_ID/gitops-example-dev.git --service promote-demo
+```
+
+## Generate a GitHub token
+
+You will need a GitHub token with repository access in order to merge Pull Requests. 
+
+## Standalone case
+
+Edit the four files in the 'templates' directory.
+
+- In `automerge-task.yaml` replace `YOUR_DOCKER_HUB_ID` with your DockerHub id.
+- In `git-resource.yaml` replace `YOUR_GITHUB_ID` with your GitHub id.
+- In `github-secret.yaml` replace `[your-github-token-with-repo-access]` with your base64-encoded token. Run this command to get the right string:
+
+```sh
+echo -n [your-token] | base64
```

Comment:
nit: You could have used stringData rather than data in the yaml file, then I do not believe you need to do the base64 conversion in advance.  (I'm pretty sure if your yaml states stringData, kube does the base64 conversion when the yaml is applied - up to you whether you test this and change)

Reply:
done